### ISSUE TITLE
Decomposition - cudnn_batch_norm vs native_batch_norm

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#611]
For few of the TIMM models, there are accuracy differences when I use AOT Autograd. On further debugging, I found that removing `cudnn_batch_norm` and `cudnn_batch_norm_backward` resolves the issue. 

I am able to repro it, but in somewhat unusual way. I have incpetion model. In the first version, all the batch norm ops are `cudnn_batch_norm` ops. In the second version, I have replaced the very first `cudnn_batch_norm` with `native_batch_norm`. With this, I am able to repro the accuracy difference.

The zip file is here

[inception.tar.gz](https://github.com/pytorch/torchdynamo/files/9145065/inception.tar.gz)

The file - `inception_fail.py` compares the above two versions of the model and shows that the output tensors deviate a lot.

**Comment by anijain2305** [comment]
cc @Chillee @desertfire

**Comment by anijain2305** [comment]
@ptrblck identified the issue here that TF32 matmuls are amplifying the noise coming from the first matmul. There are some minor differences between native and cudnn version of batch_norm, which in isolation are ok, but when amplified through TF32 matmuls cause issues.

So, a better way is to either use FP32 for accuracy comparisons. This might also not be enough, as I see in `inception_v3`, where the accuracy failure is now in the gradient value (compared to forward output when we were using TF32)

So, there is no other way than relaxing the accuracy constraint further. I am going to use cosine similarity for such models.
